src/utils/convert.hpp

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CONVERT_HPP_
+#define CONVERT_HPP_
+
+#include <aos/common/cloudprotocol/alerts.hpp>
+#include <aos/common/cloudprotocol/envvars.hpp>
+#include <aos/common/cloudprotocol/log.hpp>
+#include <aos/common/monitoring/monitoring.hpp>
+#include <aos/common/types.hpp>
+#include <aos/sm/networkmanager.hpp>
+
+#include <servicemanager/v4/servicemanager.grpc.pb.h>
+
+namespace aos::sm::utils {
+
+/**
+ * Converts byte array to aos.
+ *
+ * @param arr array to convert.
+ * @return const Array<uint8_t>.
+ */
+const Array<uint8_t> ConvertByteArrayToAos(const std::string& arr);
```

Comment:
Probably we have already such conversion if not it should be part of aos_core_lib_cpp.

Reply:
Nope, We have `Array<uint8_t> String::AsByteArray() const { return Array<uint8_t>(reinterpret_cast<const uint8_t*>(CStr()), Size()); }`

Decided to follow existing code pattern and use cast directly
`result.mSHA256      = {reinterpret_cast<const uint8_t*>(val.sha256().c_str()), val.sha256().length()};`